soc/nxp/s32/s32k5/watchdog_init.c

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/arch/arm/arch.h>
+#include <zephyr/sys/util.h>
+#include <mc_me.h>
+
+/* SWT Startup Control Register */
+#define SWT_STARTUP_CR                  0x404A8000
+/* SWT Control Register bits */
+#define SWT_CR_WEN                      BIT(0)
+
+#define MCME_SWT_STARTUP_REQ            BIT(10)
+#define MCME_SWT_STARTUP_COFB_IDX       1
+#define MCME_SWT_STARTUP_PARTITION_IDX  2
+
+/*
+ * The SWT startup is enabled by default, with a timeout of 320 counter clocks.
+ * Since the watchdog driver disables the watchdog too late, use the watchdog hook
+ * to prevent resets.
+ */
+void z_arm_watchdog_init(void)
```

Comment:
@nashif for comment on this, this seems to be a weird thing that should have been removed years ago and replaced with platform hooks which is only enabled with a "secret" locally defined Kconfig which doesn't exist at a global scope. https://github.com/zephyrproject-rtos/zephyr/blob/main/arch/arm/core/cortex_m/reset.S#L189 is where this is called with the "secret" Kconfig that doesn't exist

Reply:
I've updated to not use the "secret" Kconfig